src/main/java/org/tron/p2p/connection/business/handshake/HandshakeService.java

```diff
@@ -33,6 +37,8 @@ public void processMessage(Channel channel, Message message) {
     DisconnectCode code = ChannelManager.processPeer(channel);
     if (code != DisconnectCode.NORMAL) {
       sendHelloMsg(channel, code);
+      DisconnectReason disconnectReason = getDisconnectReason(code);
+      channel.send(new P2pDisconnectMessage(disconnectReason));
```

Comment:
Both of these messages will cause disconnection. Sending one is enough. It is recommended not to change the logic here.